### PR TITLE
Integration test case problem solved.

### PR DESCRIPTION
**Pull request by jrdalpra**
@dmarkov  This PR solve the issue #64 

The problem resides at
com.amazonaws.services.dynamodbv2.model.PutItemRequest that has no
capacity value (null).

I think another bug must be created to solve this particular issue.

### CODE REVIEW

**Comment by dmarkov** [comment]
@jrdalpra Many thanks for the PR, let me find a reviewer for it

**Comment by dmarkov** [comment]
@dmzaytsev please review

**Comment by dmzaytsev** [comment]
@jrdalpra some tests failed, you can see the errors in Travis's log
fix it please

**Comment by jrdalpra** [comment]
@dmzaytsev done

**Comment by dmzaytsev** [comment]
@jrdalpra sorry, I didn't get how did you resolve the puzzle?
You just enabled the test and removed `@NotNull`

but the puzzle creator wrote

> ...should be fixed in methods AwsIteratorITCase.iteratesItems() and AwsIteratorITCase.iteratesItemsAndDeletes(). Also see RegionITCase

**Comment by jrdalpra** [comment]
@dmzaytsev The problem resides at AWS Java API. It's not returning the ConsumedCapacity object. So, to solve the IT I disable the not null check because everything works fine even without the annotation.

The "print" static method at AwsTable already threats the null problem inside it.

I think this issue problem is solved.

Another issue must threats the AWS Java api problem.

Am I wrong?

Thanks.

**Comment by jrdalpra** [comment]
@dmzaytsev I think the local instance of DynamoDB is not returning the ConsumedCapacity json attribute. That's the source of all problems.

**Comment by dmzaytsev** [comment]
@jrdalpra well...
I guess it doesn't requests for ConsumedCapacity that's why it doesn't returns. It's not a bug, just by design.

I just google it in 1 minute
http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/dynamodbv2/model/ConsumedCapacity.html
> ConsumedCapacity is only returned if the request asked for it. 

so I think you should another use #putItem() method inside https://github.com/jcabi/jcabi-dynamo/blob/master/src/main/java/com/jcabi/dynamo/AwsTable.java

the method that allows to request `ConsumedCapacity`
see http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_PutItem.html#DDB-PutItem-request-ReturnConsumedCapacity

There is the puzzle subject I guess. @yegor256 what do you think?

Anyway you should create PR if you already have a decision.
Otherwise if you have a question you should ask the architect of the project, @yegor256 is architect here. Or you can create a new ticket if you find a bug or something else that blocks your job. Please check out the policy http://at.teamed.io/policy.html
Code reviewer just check your design and the code style.

**Comment by jrdalpra** [comment]
@dmzaytsev The AwsTable class already asks for ConsumedCapacity. So, I think the problem resides at AWS Java Api or, most probably, at Local Dynamo Server used to IT tests.

So, I'll ask for @dmarkov remove me from this issue and I'll create a new issue reporting the bug about retrieving the ConsumedCapacity.

**Comment by jrdalpra** [comment]
@dmzaytsev @yegor256 @dmarkov 

Found the problem. I updated the jcabi-dynamodb-maven-plugin version and returned the @NotNull at AwsTable.print(ConsumedCapacity) method.

IT is running like a charm.

**Comment by dmzaytsev** [comment]
@jrdalpra 

>IT is running like a charm.

CI servers don't think so

**Comment by jrdalpra** [comment]
See the build running at my machine.

[build-result.txt](https://github.com/jcabi/jcabi-dynamo/files/61463/build-result.txt)

**Comment by dmzaytsev** [comment]
@jrdalpra branches will merge here, not on your machine

**Comment by jrdalpra** [comment]
@dmzaytsev @dmarkov The problem is not at this project.

The problem resides at Local DynamoDB Server (used for tests)

http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Tools.DynamoDBLocal.html says

Consumed capacity units are not tracked. In API responses, nulls are returned instead of capacity units.

So what I must do now?

The puzzle is solved, problem comes from outside project.

Thanks.

**Comment by dmzaytsev** [comment]
@yegor256 please advise to the question above

**Comment by jrdalpra** [comment]
@dmzaytsev @yegor256 what's the PR status?

**Comment by jrdalpra** [comment]
@dmzaytsev @yegor256 any news?

**Comment by jrdalpra** [comment]
@dmzaytsev @yegor256 Can you give me a feedback? I want to get more tasks and I have de limit of 2. Thanks.

File: src/main/java/com/jcabi/dynamo/AwsTable.java [hunk 1]
```diff
@@ -190,7 +190,7 @@ public String name() {
      */
     @NotNull(message = "String cannot be null")
     public static String print(
-        @NotNull(message = "attribute capacity cannot be null")
+//        @NotNull(message = "attribute capacity cannot be null")
```
Comment (yegor256) [inline src/main/java/com/jcabi/dynamo/AwsTable.java]: @jrdalpra we're using Git, we don't need comments. Just remove the line

**Comment by yegor256** [comment]
@jrdalpra see my comment above

File: src/main/java/com/jcabi/dynamo/AwsTable.java (hunk 1 above)
Comment (jrdalpra) [inline src/main/java/com/jcabi/dynamo/AwsTable.java]: @yegor256 done

The qulice is not getting this kind of restriction.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by dmarkov** [comment]
@dmzaytsev Thanks a lot, I just topped your account for **21 mins**, transaction ID `72542066`, total time was 97 hours and 27 mins

extra minutes for review comments (c=6)

+21 added to your rating, current score is: [+996](http://www.netbout.com/b/35166?open=rating)

**Comment by dmarkov** [comment]
@rultor deploy

**Comment by jrdalpra** [comment]
@dmarkov what about me? Task is done but it's not closed. Thanks.

**Comment by dmarkov** [comment]
> @dmarkov what about me? Task is done but it's not closed. Thanks.

@jrdalpra looks like a bug in GitHub... just close it yourself

**Comment by jrdalpra** [comment]
@dmarkov @yegor256 @dmzaytsev PR merged and task still open at netbout. How can we solve it?

**Comment by jrdalpra** [comment]
closes jcabi/jcabi-dynamo#64

**Pull request merged**